library/psa_crypto_cipher.c

```diff
@@ -531,22 +535,17 @@ static psa_status_t cipher_decrypt( const psa_key_attributes_t *attributes,
     status = cipher_update( &operation, input + operation.iv_length,
                             input_length - operation.iv_length,
                             output, output_size, &olength );
-    accumulated_length = olength;
     if( status != PSA_SUCCESS )
         goto exit;
 
-    if( output_size < accumulated_length )
-    {
-        status = PSA_ERROR_BUFFER_TOO_SMALL;
-        goto exit;
-    }
+    accumulated_length = olength;
 
     status = cipher_finish( &operation, output + accumulated_length,
                             output_size - accumulated_length, &olength );
-    accumulated_length += olength;
     if( status != PSA_SUCCESS )
         goto exit;
 
+    accumulated_length += olength;
```

Comment:
```suggestion
    *output_length = accumulated_length + olength;
```
That's probably a bit better.